src/analyzer/analyzermanager.cpp

```diff
@@ -0,0 +1,298 @@
+#include "analyzer/analyzermanager.h"
+
+#include <typeinfo>
+#include <QThread>
+
+#include <QtDebug>
+#include <QMutexLocker>
+#include <QListIterator>
+
+#include "library/trackcollection.h"
+#include "mixer/playerinfo.h"
+#include "track/track.h"
+#include "util/compatibility.h"
+#include "util/event.h"
+#include "util/timer.h"
+#include "util/trace.h"
+
+AnalyzerManager* AnalyzerManager::m_pAnalyzerManager = NULL;
+
+//static
+AnalyzerManager& AnalyzerManager::getInstance(UserSettingsPointer pConfig) {
+    if (!m_pAnalyzerManager) {
+        // There exists only one UserSettingsPointer in the app, so it doens't matter
+        // if we only assign it once.
+        m_pAnalyzerManager = new AnalyzerManager(pConfig);
+    }
+    int maxThreads = m_pAnalyzerManager->m_pConfig->getValue<int>(ConfigKey("[Library]", "MaxAnalysisThreads"));
+    int ideal = QThread::idealThreadCount();
+    if (QThread::idealThreadCount() < 1) {
+        ideal = 1;
+    }
+    if (maxThreads <= 0 || maxThreads > 32) {
+        //Assume the value is incorrect, so fix it.
+        maxThreads = ideal;
+    }
+    m_pAnalyzerManager->m_MaxThreads = maxThreads;
+    return *m_pAnalyzerManager;
+}
+
+AnalyzerManager::AnalyzerManager(UserSettingsPointer pConfig) :
+    m_pConfig(pConfig),
+    m_nextWorkerId(0),
+    m_batchTrackQueue(),
+    m_prioTrackQueue(),
+    m_backgroundWorkers(),
+    m_foregroundWorkers(),
+    m_pausedWorkers() {
+}
+
+AnalyzerManager::~AnalyzerManager() {
+    stop(true);
+}
+
+bool AnalyzerManager::isActive(bool includeForeground) {
```

Comment:
isActive(true) is hard to read at the caller. 
how about add two calls? isActive() and isBackgroundWorkerActive()